<launch>

<node name="rviz" pkg="rviz" type="rviz" args="-d $(find trajectory_generation)/cfg/2D_map.rviz"/>

<node pkg="trajectory_generation" name="trajectory_generation" type="trajectory_generation" output="screen" required = "true" >
    <param name="trajectory_generator/map_x_size" value="30.0"/>
    <param name="trajectory_generator/map_y_size" value="18.0"/>  <!-- 仿真环境参数-->
<!--     <param name="trajectory_generator/map_y_size" value="30.0"/> -->
    <param name="trajectory_generator/map_z_size" value="2.0"/>
    <param name="trajectory_generator/search_height_min" value="-0.3"/>  <!-- 仿真环境参数-->
<!--     <param name="trajectory_generator/search_height_min" value="0.1"/> -->
    <param name="trajectory_generator/search_height_max" value="1.2"/>
    <param name="trajectory_generator/search_radius" value="6.0"/>
    <param name="trajectory_generator/obstacle_swell_flag" value="true"/>
    <param name="trajectory_generator/robot_radius_dash" value="0.35"/>
    <param name="trajectory_generator/robot_radius" value="0.35"/>
    <param name="trajectory_generator/isxtl" value="false"/>
    <param name="trajectory_generator/xtl_flag" value="false"/>
    <param name="trajectory_generator/map_resolution" value="0.1"/>

    <param name="trajectory_generator/reference_v_max" value="3.5"/>
    <param name="trajectory_generator/reference_a_max" value="10.0"/>  <!--xtl:8.0  normal 11.0-->>
    <param name="trajectory_generator/reference_w_max" value="4.0"/>
    <param name="trajectory_generator/reference_desire_speed" value="2.5"/>  <!--xtl 1.8，正常环境极限速度2.8，可能实际更难达到-->>
    <param name="trajectory_generator/reference_desire_speedxtl" value="2.2"/>
    <param name="trajectory_generator/reference_axtl_max" value="2.0"/>
    <param name="trajectory_generator/reference_wxtl_max" value="2.0"/>

    <param name="trajectory_generator/map_lower_point_x" value="0.0"/> <!--这个参数是很多年前的东西了不知道还有没有用了懒得看了-->
    <param name="trajectory_generator/map_lower_point_y" value="0.0"/> <!--我们一般都把地图起点设置成红方补弹区角点因此不存在水平偏置-->
    <param name="trajectory_generator/map_lower_point_z" value="0.0"/>

    <param name="trajectory_generator/height_bias" value="-0.5"/>  <!-- 地图高度偏置参数-->
    <param name="trajectory_generator/height_interval" value="2.0"/> <!-- 地图高度间隔参数,高度地图所有像素的高度间隔-->
    <param name="trajectory_generator/height_threshold" value="0.1"/> <!-- 地图高度阈值参数,障碍物点云高度检测阈值-->
    <param name="trajectory_generator/height_sencond_high_threshold" value="0.3"/> <!-- 桥梁地图高度阈值参数,这个是顶部高度阈值-->

<!--     <param name="trajectory_generator/occ_file_path" type="string" value="/home/hitcrt/sentry_ws/src/sentry_planning/trajectory_generation/map/occ506.png"/> -->
<!--     <param name="trajectory_generator/bev_file_path" type="string" value="/home/hitcrt/sentry_ws/src/sentry_planning/trajectory_generation/map/bev506.png"/> -->
<!--     <param name="trajectory_generator/distance_map_file_path" type="string" value="/home/hitcrt/sentry_ws/src/sentry_planning/trajectory_generation/map/occtopo506.png"/> -->

    <param name="trajectory_generator/occ_file_path" type="string" value="$(find trajectory_generation)/map/occ2024low.png"/>
    <param name="trajectory_generator/bev_file_path" type="string" value="$(find trajectory_generation)/map/bev2024low.png"/>
    <param name="trajectory_generator/distance_map_file_path" type="string" value="$(find trajectory_generation)/map/occtopo2024low.png"/>

</node>

<node pkg="waypoint_generator" name="waypoint_generator" type="waypoint_generator" output="screen">
    <remap from="~goal" to="/goal"/>
    <param name="waypoint_type" value="manual-lonely-waypoint"/>
</node>

</launch>
